distributed/protocol.py

```diff
@@ -64,35 +62,77 @@
     default_compression = 'lz4'
 
 
+def _extract_big_bytes(x, big, path=()):
+    if type(x) is dict:
+        for k, v in x.items():
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
+                big[path + (k,)] = v
+                del x[k]
+    elif type(x) is list:
+        for k, v in enumerate(x):
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
+                big[path + (k,)] = v
+                x[k] = None
+
+
 def dumps(msg):
```

Comment:
Hrm, that's a good point, this is fairly far buried within other calls, like rpc.  

In practice the channels that receive large bytestrings are scatter, gather, worker-worker communication and occasionally update-graph.  But this is probably not something you should dive into at the moment.

Alternative question: how expensive is this change?  What is the overhead on dumping small messages?